pkg/teeth/sql/publisher_source.sql

```diff
@@ -0,0 +1,42 @@
+SELECT
+  pull_request_events.delivery_id,
+  delivery_events.pull_request_id,
+  html_url AS pull_request_html_url,
+  delivery_events.received,
+  logs_uri,
+  head_sha
+FROM
+  `%s` AS pull_request_events
```

Comment:
Why not use a WITH statement?

Reply:
I rarely use `WITH`  unless the `WITH` table is reused and **NOT** joined on. This is not the case here; the nested queries are for the most part joined apart from the subquery in the `invocation_comment_status` table.

Here's an example I found internally where WITH is useful.

```
WITH w AS (
  SELECT MAX(last_batch_run) FROM batch_runs
)
SELECT * 
FROM table1
 JOIN table 2 USING(id)
 JOIN table 3 USING(id)
WHERE table1.batch_timestamp = (SELECT * FROM w)
  AND table2.batch_timestamp = (SELECT * FROM w)
  AND table3.batch_timestamp = (SELECT * FROM w)
```

Reply:
Are you imagining that to just be a view of some sort? I assumed you were injecting a subquery.

Reply:
Imagining what to be a view?

Reply:
`%s` AS pull_request_events

The %s string replacement, is it going to be a subquery embedded inside or a view?

Reply:
That's just the BQ table name that needs to be specified by the job config. 

e.g.,
`github-metrics-aggregator.github-metrics.pull_request_events AS pull_request_events`

Reply:
I see. You may want to use the TVF versions of those queries. For `abcxyz` repo, the cost isn't significant, but for a larger repo it could be. The cost of the `unique_events` view `pull_request_events` and others count on is a scan of the entire table. I used TVFs to take advantage of the clustering and partitioning in `raw_events`

Reply:
Ack. With how things are set up now, we can configure which BQ source to use whether it be a TVF, routine, view, or other table down the line. You can make the case for all other tables here but let's get some concrete numbers to back up such optimizations before doing them.